### ISSUE TITLE
Embedding video from vimeo error

### ISSUE THREAD

**Issue by danFbach** [nopSolutions/nopCommerce#7299]
nopCommerce version: 4.6

Vimeo is pretty aggressive in blocking "bot" traffic. It blacklists many datacenter's IP ranges, it fingerprints http clients to try to detect bots among many other methods. If you try to add a **vimeo** video to a product using Multimedia video tab, you will get either a `429 (too many requests)` response or `401 (Unauthorized)`. This may not **always** be true, but a simple google search will pull up many reddit and stackoverflow threads about this issue.

I finally got a response from vimeo with a solution using their `oembed API` to perform this check.

I have changed the PingVideoUrlAsync method:
```cs
protected virtual async Task PingVideoUrlAsync(string videoUrl)
{
    var path = videoUrl.StartsWith("/") ? $"{_webHelper.GetStoreLocation()}{videoUrl.TrimStart('/')}" : videoUrl;

    if (path.Contains("vimeo.com"))
        path = "https://vimeo.com/api/oembed.json?url=" + WebUtility.UrlEncode(path);

    var client = _httpClientFactory.CreateClient(NopHttpDefaults.DefaultHttpClient);
    await client.GetStringAsync(path);
}
```
and I can add vimeo videos again.

see [vimeo docs](https://developer.vimeo.com/api/oembed/videos) for reference to oembed api.

The suggested approach looks like a workaround. Let's find a better way to allow such video

**Comment by AndreiMaz** [comment]
The suggested approach looks like a workaround. Let's find a better way to allow such video

**Comment by DmitriyKulagin** [comment]
Closed #7299
